### PR TITLE
Add rule test and create helper for common in tests

### PR DESCRIPTION
**Pull request by gartz**
It contains a few tests for rules, and it changes the tests to use a helper for the common before and before each actions regarding the browser history API events.

### CODE REVIEW

File: test/route-beutifyLocation.js [hunk 1]
```diff
@@ -1,34 +1,9 @@
-var PushStateTree = require('../src/pushStateTree');
+const PushStateTree = require('../src/pushStateTree');
+import cleanHistoryAPI from './helper/cleanHistoryAPI';
 
 describe('PushStateTree beutifyLocation should', function() {
```
Comment (AntonioSchmidt) [inline test/route-beutifyLocation.js:4]: nab: as the 'should'  have been removed from describe and put in the 'it' function in route-basePath.js, wouldn't be better if the other files followed the same pattern?

File: test/rule.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const PushStateTree = require('../src/pushStateTree');
```
Comment (AntonioSchmidt) [inline test/rule.js:1]: most of the tests in this file uses the createRule function which is tested in route-CreateRules.js.

File: test/rule.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  });
+
+  it('should be created by PushStateTree-router instance', function () {
+    let rule = pst.createRule();
+    expect(rule).to.be.defined;
+  });
+
+  it('should be a HTMLElement instance', function () {
+    let rule = pst.createRule();
+    expect(rule).to.be.instanceof(HTMLElement);
```
Comment (AntonioSchmidt) [inline test/rule.js]: route-CreateRules.js has the same tests

File: test/route-beutifyLocation.js (hunk 1 above)
Comment (gartz) [inline test/route-beutifyLocation.js:4]: I agree, but not in this PR.

**Pull request merged**